bittide/tests/Tests/StabilityChecker.hs

```diff
@@ -0,0 +1,64 @@
+-- SPDX-FileCopyrightText: 2022 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE OverloadedStrings #-}
+
+module Tests.StabilityChecker where
+
+import Prelude ((^))
+import Clash.Prelude hiding ((^))
+import Test.Tasty
+import qualified Hedgehog.Gen as Gen
+import Hedgehog
+import Bittide.ClockControl.StabilityChecker
+import Data.Proxy
+import Tests.Shared
+import qualified Hedgehog.Range as Range
+import Clash.Hedgehog.Sized.Unsigned (genUnsigned)
+import Test.Tasty.Hedgehog
+
+stabilityGroup :: TestTree
+stabilityGroup = testGroup "StabilityChecker group"
+  [ testPropertyNamed "trackedTargetDeviation acts the same as its golden reference"
+    "trackedTargetDeviationTest" trackedTargetDeviationTest]
+
+mkTrackedTargetDeviation :: Integer -> Integer -> StabilityStrategy
+mkTrackedTargetDeviation m cs =
+  case (someNatVal m, someNatVal (cs - 1)) of
+    (Just (SomeNat margin), Just (SomeNat cyclesStable)) ->
+      TrackedTargetDeviation (snatProxy margin) (succSNat $ snatProxy cyclesStable)
+    _ -> error $
+      "mkTrackedTargetDeviation: ( 0 <= margin) and (1 <= cyclesStable) not met: margin = "
+      <> show m <> ", cyclesStable = " <> show cs <> "."
+
+trackedTargetDeviationTest :: Property
+trackedTargetDeviationTest = property $ do
+  dataCountBits <- forAll $ Gen.integral $ Range.linear 2 63
```

Comment:
```suggestion
  dataCountBits <- forAll $ Gen.integral $ Range.linear 2 128
```

Cross the machine limit. (Not that we're actually going to use >64 bits to represent data count but you know..)